### DISCUSSION TITLE
Ability to change text colour using config.text_color

### DISCUSSION THREAD

**Discussion by PRU1** [ManimCommunity/manim#2790] [Suggestions and Proposals]
## Description of proposed feature

To change the background colour, `config.background_color=COLOUR` can be used. Adding support to change the text colour would be convenient when changing the background colour, and the default white text doesn't show well.

## How can the new feature be used?

Maybe a command like `config.text_color=COLOUR`

**Comment by markromanmiller** [comment]
I see the need - it would be great to set the color of a type of Mobject exactly once in the code or stylesheet configuration.

However, the dev team has decided that config is not the right place for background_color and we want to deprecate it soon. There isn't yet a solidified plan for a "stylesheet" type interface, though I agree I'd like one.

Last time this came up on Discord, the two kinds of workarounds were subclassing Tex with different defaults or defining a partial function that modifies kwargs if a value is unset.

If you want to take on a larger project, feel free to suggest some alternatives and start developing one of them. I'd be happy to have stylesheets in Manim.

**Comment by cobordism** [comment]
One positive thing about setting so many attributes as keyword arguments is that you can easily make your own version of a mobject constructor with different defaults:


```
def myText(*args, color=PURPLE, **kwargs):
    return Text(*args, color=color, **kwargs)
```
and then use myText instead of Text in you scene.

**Comment by kilacoda-old** [comment]
One possibly better solution might be to add a new config property `config.default_mobject_color` which would replace the currently hardcoded color value of `WHITE` in `Mobject.__init__`:

https://github.com/ManimCommunity/manim/blob/dbb62129b620d17d1828aefd5c6e1ade3d9e0c9e/manim/mobject/mobject.py#L67

**Comment by hydrobeam** [comment]
I think I'm a fan of the `config.default_mobject_color` idea, but should it also override all colours for mobjects? There are quite a few which have colourful defaults (RED for `Circle`). One way of doing it is to remove all default colours and have everything be `WHITE` / `default_mobject_color`
